### ISSUE TITLE
Use StackExchange.Redis with external libraries loaded fails in timeout (dependency injection)

### ISSUE THREAD

**Issue by ThatDeveloper** [StackExchange/StackExchange.Redis#611]
In my newest project I implemented a Plugin architecure where I load external .dll's and add them to the assembly context.

I am using it with MVC (dotNetCore) and everything works except StackExchange.Redis.
If I use it on the Main project (loader) it works, but if I use it in the library (dependency injection) I get the instance but as soon as I get or set (or whatever) I get a Timeout.

As it works in the main project i am very sure it has a problem with the assmebly. But both have the same version of StackExchange.Redis and both have all the dependencies.

Software:
Visual Studio 2017
dotNetCore 1.1 (newest)
StackExchange.Redis 1.2.1 (newest)
Windows / Linux

Exception:

```Data: {System.Collections.ListDictionaryInternal}
    HResult: -2146233083
    HelpLink: "https://github.com/StackExchange/StackExchange.Redis/tree/master/Docs/Timeouts.md"
    InnerException: null
    Message: "Timeout performing EVAL, inst: 0, queue: 1, qu: 1, qs: 0, qc: 0, wr: 0, wq: 1, in: 0, ar: 0, clientName: DEVELOP01 (Please take a look at this article for some common client-side issues that can cause timeouts: https://github.com/StackExchange/StackExchange.Redis/tree/master/Docs/Timeouts.md)"
    Source: "StackExchange.Redis.StrongName"
    StackTrace: "   at StackExchange.Redis.ConnectionMultiplexer.ExecuteSyncImpl[T](Message message, ResultProcessor`1 processor, ServerEndPoint server)\r\n   at StackExchange.Redis.RedisBase.ExecuteSync[T](Message message, ResultProcessor`1 processor, ServerEndPoint server)\r\n   at StackExchange.Redis.RedisDatabase.ScriptEvaluate(String script, RedisKey[] keys, RedisValue[] values, CommandFlags flags)\r\n   at Microsoft.Extensions.Caching.Redis.RedisCache.Set(String key, Byte[] value, DistributedCacheEntryOptions options)"
    TargetSite: {T ExecuteSyncImpl[T](StackExchange.Redis.Message, StackExchange.Redis.ResultProcessor`1[T], `StackExchange.Redis.ServerEndPoint)}```

**Comment by benaadams** [comment]
Timeout doc is here: https://stackexchange.github.io/StackExchange.Redis/Timeouts

**Comment by NickCraver** [comment]
Are you still seeing this in latest? (e.g. 1.2.6 or above?) We fixed several socket issues present in 1.2.1 through 1.2.5 due to a build regression and also added a bunch of information to the debug string to let us figure out why timeouts are occuring.

**Comment by mgravell** [comment]
Closing; please see #871
